runtime/autoload/provider/node.vim

```diff
@@ -71,13 +71,15 @@ function! provider#node#Detect() abort
     let yarn_opts = deepcopy(s:NodeHandler)
     let yarn_opts.entry_point = '/node_modules/neovim/bin/cli.js'
     " `yarn global dir` is slow (> 250ms), try the default path first
-    " XXX: The following code is not portable
     " https://github.com/yarnpkg/yarn/issues/2049#issuecomment-263183768
     if has('unix')
-      let yarn_default_path = $HOME . '/.config/yarn/global/' . yarn_opts.entry_point
-      if filereadable(yarn_default_path)
-        return [yarn_default_path, '']
-      endif
+      let yarn_config_dir = '/.config/yarn/'
+    elseif has('win32')
+      let yarn_config_dir = '/AppData/Local/Yarn/Data/'
+    endif
+    let yarn_default_path = $HOME . yarn_config_dir . '/global/' . yarn_opts.entry_point
```

Comment:
It seems that `$LOCALAPPDATA` should be used on Windows. `$HOME` may not be always present.

Reply:
$HOME should always be present given Microsoft's focus on making Windows PowerShell 7 similar to unix systems, and away from the old CMD. It's basically just `C:\\Users\\UserName`. If we use `$LOCALAPPDATA`, then it has to be `$ENV:LOCALAPPDATA`, which then will add more code and might cause confusion.

Reply:
Oh, actually `$HOME` has some special handling in Vim script on Windows. Never mind then.